### PR TITLE
 fr-115_Toolbar

### PR DESCRIPTION
**Pull request by OlegMoshkovich**
- Subtract the toolbar from the Cad view
- Add new logo
- Add a new favicon
- Move logo to the bottom left
- Create base button group

https://olegmoshkovich.github.io/Share/share/v/p/index.ifc

### CODE REVIEW

**Comment by OlegMoshkovich** [comment]
@pablo-mayrgundter PTAL!

**Comment by OlegMoshkovich** [comment]
@pablo-mayrgundter  
PTAL!

File: src/Components/AboutPanel.jsx [hunk 1]
```diff
@@ -11,13 +12,15 @@ import About from '../assets/3D/attention.svg'
  * @return {Object} The AboutControl react component.
  */
 export default function AboutControl({offsetTop}) {
-  const [open, setOpen]=React.useState(true)
+  const [open, setOpen]=useState(true)
```
Comment (pablo-mayrgundter) [inline src/Components/AboutPanel.jsx:15]: const [open,setOpen] = useState(true)

File: src/Components/AboutPanel.jsx [hunk 2]
```diff
@@ -11,13 +12,15 @@ import About from '../assets/3D/attention.svg'
...
   const classes = useStyles()
   return (
     <div >
-      <Typography className = {classes.about} onClick={() => {
-        setOpen(!open)
-      }}>About</Typography>
+      <Tooltip title="About" placement="top">
+        <IconButton onClick = {()=>setOpen(!open)}>
```
Comment (pablo-mayrgundter) [inline src/Components/AboutPanel.jsx:20]:
```
 <IconButton onClick={() => setOpen(!open)}>
```

File: src/Components/AboutPanel.jsx [hunk 3]
```diff
@@ -11,13 +12,15 @@ import About from '../assets/3D/attention.svg'
...
+          <About className = {classes.icon}/>
```
Comment (pablo-mayrgundter) [inline src/Components/AboutPanel.jsx:21]:
```
<About className={classes.icon}/>
```

File: src/Components/BaseGroup.jsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import Settings from './Settings'
+import AboutControl from './AboutPanel'
+import Open from '../assets/Icons/Open.svg'
+
+
+/** Base group contains Settings, ModelUpload, About
+ * @param {Function} fileOpen opens a file dialog for new model upload
+ * @param {Number} offsetTop
+ * @return {Object} React component.
+ */
```
Comment (pablo-mayrgundter) [inline src/Components/BaseGroup.jsx:14]:
```
/**
 * Base group contains Settings, ModelUpload, About
 * @param {Function} fileOpen opens a file dialog for new model upload
 * @param {Number} offsetTop
 * @return {Object} React component.
 */
```

File: src/Components/ItemPanel.jsx [hunk 5]
```diff
@@ -1,29 +1,43 @@
 import React, {useState} from 'react'
 import {makeStyles} from '@mui/styles'
```
Comment (pablo-mayrgundter) [inline src/Components/ItemPanel.jsx:2]: Please move this down to after default imports for @mui

File: src/Components/ItemPanel.jsx [hunk 6]
```diff
@@ -1,29 +1,43 @@
...
  * @param {Object} model IFC model
  * @param {Object} element The currently selected IFC element
  * @param {Number} topOffset Screen offset position
  * @return {Object} The ItemPanelButton react component
  */
-export default function ItemPanelButton({model, element, topOffset}) {
+export default function ItemPanelButton({model, element, topOffset, toggle, open}) {
   const [showItemPanel, setShowItemPanel] = useState(false)
   const classes = useStyles({topOffset: topOffset})
+  console.log('open', open)
```
Comment (pablo-mayrgundter) [inline src/Components/ItemPanel.jsx:21]: remove

File: src/Components/ItemPanel.jsx [hunk 7]
```diff
@@ -1,29 +1,43 @@
...
- * Container for ItemProperties.  ItemProperties is wrapped in an
+ * Container for ItemProperties. ItemProperties is wrapped in an
  * ItemPropertiesDrawer to toggle hiding.
  * @param {Object} model IFC model
  * @param {Object} element The currently selected IFC element
  * @param {Number} topOffset Screen offset position
  * @return {Object} The ItemPanelButton react component
  */
-export default function ItemPanelButton({model, element, topOffset}) {
+export default function ItemPanelButton({model, element, topOffset, toggle, open}) {
```
Comment (pablo-mayrgundter) [inline src/Components/ItemPanel.jsx:18]: add @param docs for new params, and I think better names.. it's not clear what's being toggled or opened.. this component or something from the caller?

File: src/Components/SearchBar.jsx [hunk 8]
```diff
@@ -67,15 +68,16 @@ export default function SearchBar({onClickMenuCb, isOpen}) {
         className={classes.iconButton}
         aria-label='menu'
         onClick={onClickMenuCb}>
-        <Hamburger className={classes.icon}/>
+        {showNavPanel?<Close className={classes.icon}/>:<Hamburger className={classes.icon}/>}
```
Comment (pablo-mayrgundter) [inline src/Components/SearchBar.jsx:71]:
```
{showNavPanel ? <Close className={classes.icon}/> : <Hamburger className={classes.icon}/>}
```

File: src/Containers/CadView.jsx [hunk 9]
```diff
@@ -348,11 +345,11 @@ export default function CadView({
           {showSearchBar && (
             <SearchBar
               onClickMenuCb={() => setShowNavPanel(!showNavPanel)}
+              showNavPanel = {showNavPanel}
```
Comment (pablo-mayrgundter) [inline src/Containers/CadView.jsx:348]:
```
showNavPanel={showNavPanel}
```

File: src/Containers/CadView.jsx [hunk 10]
```diff
@@ -366,23 +363,26 @@ export default function CadView({
...
-            placeCutPlane={()=>placeCutPlane()}
-            unSelectItem={()=>unSelectItems()}
-            toggleShortCutsPanel={()=>setShowShortCuts(!showShortCuts)}/>
-        </div>
-        <div className={classes.iconGroup}>
-          <IconGroup
+        <ItemPanelButton
+          model={model}
+          element={selectedElement}
+          open = {showItemPanel}
```
Comment (pablo-mayrgundter) [inline src/Containers/CadView.jsx:369]:
```
open={showItemPanel}
```

File: src/Containers/CadView.jsx [hunk 11]
```diff
@@ -366,23 +363,26 @@ export default function CadView({
...
+          toggle={()=>setShowItemPanel(!showItemPanel)}
```
Comment (pablo-mayrgundter) [inline src/Containers/CadView.jsx:370]: Next lines as well, spaces around arrow
```
() => ...
```

File: src/Containers/CadView.jsx [hunk 12]
```diff
@@ -366,23 +363,26 @@ export default function CadView({
...
+          topOffset={PANEL_TOP}
+          placeCutPlane={()=>placeCutPlane()}
+          unSelectItem={()=>unSelectItems()}
+          toggleShortCutsPanel={()=>setShowShortCuts(!showShortCuts)}/>
+        <div className={showItemPanel?classes.iconGroupOpen:classes.iconGroup}>
```
Comment (pablo-mayrgundter) [inline src/Containers/CadView.jsx:375]: spaces around operators

File: src/Containers/CadView.jsx [hunk 13]
```diff
@@ -366,23 +363,26 @@ export default function CadView({
...
+          <OperationsGroup
             placeCutPlane={()=>placeCutPlane()}
```
Comment (pablo-mayrgundter) [inline src/Containers/CadView.jsx:377]: spaces around arrows

File: src/Containers/CadView.jsx [hunk 14]
```diff
@@ -366,23 +363,26 @@ export default function CadView({
...
             unSelectItem={()=>unSelectItems()}
             toggleShortCutsPanel={()=>setShowShortCuts(!showShortCuts)}
           />
         </div>
+        <LogoDark className = {classes.logo}/>
```
Comment (pablo-mayrgundter) [inline src/Containers/CadView.jsx:382]: no spaces around =, and on next line spaces around ops

**Review by pablo-mayrgundter — CHANGES_REQUESTED** [review]
Can't wait :). Mostly just spaces

**Comment by OlegMoshkovich** [comment]
PTAL!

File: src/Components/ItemPanel.jsx [hunk 15]
```diff
@@ -13,29 +12,27 @@ import Close from '../assets/Icons/Close.svg'
  * @param {Object} model IFC model
  * @param {Object} element The currently selected IFC element
  * @param {Number} topOffset Screen offset position
+ * @param {function} setShowitempanel toggles the state of the showItemPanel in cadView
  * @return {Object} The ItemPanelButton react component
  */
-export default function ItemPanelButton({model, element, topOffset, toggle, open}) {
+export default function ItemPanelButton({model, element, topOffset, setShowitempanel}) {
```
Comment (pablo-mayrgundter) [inline src/Components/ItemPanel.jsx:18]: use the same casing as the original var, so setShowItemPanel

File: src/Components/ItemPanel.jsx [hunk 16]
```diff
@@ -13,29 +12,27 @@ import Close from '../assets/Icons/Close.svg'
...
   const [showItemPanel, setShowItemPanel] = useState(false)
   const classes = useStyles({topOffset: topOffset})
-  console.log('open', open)
 
   return (
-    <div className={showItemPanel?classes.toggleButtonOpen:classes.toggleButton}>
+    <div className={classes.toggleButton}>
       <IconButton onClick={() => {
-        toggle()
+        setShowitempanel()
```
Comment (pablo-mayrgundter) [inline src/Components/ItemPanel.jsx:25]: Ah, looks like there's a name conflict.. let's discuss

File: src/Components/ItemProperties.jsx [hunk 17]
```diff
@@ -32,13 +32,18 @@ export default function ItemProperties({model, element}) {
...
+      <>
+        <h2 className={classes.sectionTitle}>Properties</h2>
+        {propTable || 'Loading...'}
+        <h2 className={classes.sectionTitle}>
+          Property Sets
+          <Toggle onChange={() => setExpandAll(!expandAll)} />
+        </h2>
+        {psetsList || 'Loading...'}
+      </>
+      }
```
[the remark this answers is not in the document]
Comment (pablo-mayrgundter) [reply src/Components/ItemProperties.jsx:47]: Almost, move operators to front of line like in my example

File: src/Components/ItemPropertiesDrawer.jsx [hunk 18]
```diff
@@ -29,6 +30,7 @@ export default function ItemPropertiesDrawer({
       <div className={classes.drawerContainer}>
         <div className={classes.headerWrapper} >
           <div className={classes.title}>{title}</div>
+          <Close className={classes.close} onClick = {onClose}/>
```
Comment (pablo-mayrgundter) [inline src/Components/ItemPropertiesDrawer.jsx:33]: space

File: src/Components/ShortcutsPanel.jsx [hunk 19]
```diff
@@ -14,13 +15,17 @@ export default function ShortcutsControl({offsetTop}) {
...
+      <IconButton onClick={() => {
         setOpen(!open)
-      }} offsetTop={offsetTop}/>}
-    </IconButton>)
+      }}>
+        <Question className = {classes.icon}/>
+        {open && <ShortcutsPanel openToggle={()=>{
+          setOpen(!open)
+        }} offsetTop={offsetTop}/>}
+      </IconButton>
```
[the remark this answers is not in the document]
Comment (pablo-mayrgundter) [reply src/Components/ShortcutsPanel.jsx:32]: ops at beginning of line

**Review by pablo-mayrgundter — CHANGES_REQUESTED** [review]

**Comment by OlegMoshkovich** [comment]
PTAL!

**Pull request merged**